src/jrd/recsrc/RecordSource.h

```diff
@@ -1462,6 +1462,59 @@ namespace Jrd
 		NestConst<BoolExprNode> const m_boolean;
 	};
 
+	class TableValueFunctionScan : public RecordStream
+	{
+	protected:
+		struct Impure : public RecordSource::Impure
+		{
+			RecordBuffer* m_recordBuffer;
+			blb* m_blob;
```

Comment:
Aren't you mixing specifics of UnlistFunctionScan's impure in the base TableValueFunctionScan?